strongbox-user-management/src/main/java/org/carlspring/strongbox/users/security/SecurityTokenProvider.java

```diff
@@ -0,0 +1,128 @@
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.keys.HmacKey;
+import org.jose4j.lang.JoseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Used to get and verify security tokens. <br>
+ * This implementation based on JSON Web Token (JWT) which is RFC 7519 standard.
+ * <br>
+ * 
+ * 
+ * @author Sergey Bespalov
+ */
+@Component
+public class SecurityTokenProvider
+{
+    /**
+     * Secret key which is used to encode and verify tokens.<br>
+     * All previous tokens will be invalid, if it changed.
+     */
+    private Key key;
+
+    /**
+     * Creates {@link Key} instance using Secret string from application
+     * configuration.
+     * 
+     * @param secret
+     * @throws Exception
+     */
+    @Autowired
+    public void init(@Value("${strongbox.security.jwtSecret:secret}") String secret)
+                                                                                     throws Exception
+    {
+        key = new HmacKey(secret.getBytes("UTF-8"));
+    }
+
+    /**
+     * Generates an encrypted token.
+     * 
+     * @param subject
+     *            a Subject which is used as token base.
+     * @param claimMap
+     *            an additional Claims which will also present in token.
+     * @param expire
+     *            token exparation date.
+     * @return encrypted token string.
+     * @throws JoseException
+     */
+    public String getToken(String subject,
+                           Map<String, String> claimMap,
+                           Date expire)
+                                        throws JoseException
+    {
+        JwtClaims claims = new JwtClaims();
+        claims.setIssuer("Strongbox");
+        claims.setGeneratedJwtId();
+        claims.setSubject(subject);
+        claimMap.entrySet().stream().forEach((e) -> {
+            claims.setClaim(e.getKey(), e.getValue());
+        });
+
+        JsonWebSignature jws = new JsonWebSignature();
+        jws.setPayload(claims.toJson());
+        jws.setKey(key);
+        jws.setDoKeyValidation(false);
+        jws.setAlgorithmHeaderValue(AlgorithmIdentifiers.HMAC_SHA256);
+
+        return jws.getCompactSerialization();
+    }
+
+    /**
+     * @param token
+     * @param targetSubject
+     * @param claimMap
+     */
+    public void verifyToken(String token,
+                            String targetSubject,
+                            Map<String, String> claimMap)
+    {
+        JwtConsumer jwtConsumer = new JwtConsumerBuilder()
+                                                          .setRequireSubject()
+                                                          .setVerificationKey(key)
+                                                          .setRelaxVerificationKeyValidation()
+                                                          .build();
+
```

Comment:
Cosmetic/trivial:
The code style is:

        JwtConsumer jwtConsumer = new JwtConsumerBuilder().setRequireSubject()
                                                          .setVerificationKey(key)
                                                          .setRelaxVerificationKeyValidation()
                                                          .build();

(the first chained method is on the same line and then all further chained methods, align to the first dot).